service/storage_proxy.cc

```diff
@@ -2163,8 +2167,22 @@ future<> storage_proxy::do_mutate(std::vector<mutation> mutations, db::consisten
 
 future<> storage_proxy::replicate_counter_from_leader(mutation m, db::consistency_level cl, tracing::trace_state_ptr tr_state,
                                                       clock_type::time_point timeout, service_permit permit) {
-    // FIXME: do not send the mutation to itself, it has already been applied (it is not incorrect to do so, though)
```

Comment:
Does this comment still apply? I suspect it does, and if so we should probably not remove it.

Reply:
Ouch, yes, it does.